orderer/consensus/etcdraft/blockpuller.go

```diff
@@ -151,72 +150,63 @@ func NewBlockFetcher(support consensus.ConsenterSupport,
 	// TODO: change this to use the new mapping of consenters in the channel config
 	// To tolerate byzantine behaviour of `f` faulty nodes, we need a total of `3f + 1` nodes.
 	// f = maxByzantineNodes, total = len(endpoints)
-	maxByzantineNodes := uint64(len(endpoints)-1) / 3
-
-	// timeout for time based shuffling
-	shuffleTimeout := shuffleTimeoutMultiplier * clusterConfig.ReplicationPullTimeout
-	shuffleTimeoutThrehold := shuffleTimeoutPercentage
+	maxByzantineNodes := (len(endpoints) - 1) / 3
 
 	fc := cluster.FetcherConfig{
-		ShuffleTimeout:               shuffleTimeout,
-		CensorshipSuspicionThreshold: time.Minute,
-		MaxByzantineNodes:            maxByzantineNodes,
 		Channel:                      support.ChannelID(),
 		TLSCert:                      der.Bytes,
 		Endpoints:                    endpoints,
 		FetchTimeout:                 clusterConfig.ReplicationPullTimeout,
+		CensorshipSuspicionThreshold: time.Duration((int64(clusterConfig.ReplicationPullTimeout) * shuffleTimeoutPercentage / 100)),
+		PeriodicalShuffleInterval:    shuffleTimeoutMultiplier * clusterConfig.ReplicationPullTimeout,
+		MaxRetries:                   uint64(clusterConfig.ReplicationMaxRetries),
+		MaxByzantineNodes:            maxByzantineNodes,
+	}
+
+	// TODO: Should this func be moved to `orderer/common/cluster` package
+	blockVerifierFunc := func(header *common.BlockHeader, metadata *common.BlockMetadata) error {
+		return verifyBlockSequence([]*common.Block{
+			{
+				Header:   header,
+				Metadata: metadata,
+			},
+		}, support.ChannelID())
 	}
 
 	bf := cluster.BlockFetcher{
-		Signer: support,
-		Dialer: stdDialer,
-		AttestationSourceFactory: func(c cluster.FetcherConfig) cluster.AttestationSource {
+		FetcherConfig:   fc,
+		LastConfigBlock: &common.Block{},
+		BlockVerifierFactory: func(block *common.Block) cluster.BlockVerifierFunc {
+			// block is a config block
+			return blockVerifierFunc
+		},
+		VerifyBlock: blockVerifierFunc,
+		AttestationSourceFactory: func(c cluster.FetcherConfig, latestConfigBlock *common.Block) cluster.AttestationSource {
+			// TODO: update FetcherConfig from latestConfigBlock
```

Comment:
I added the function `EndpointconfigFromConfigBlockV3` in the previous commit, you can use it.